Allow use of HAAS_ENDPOINT env for haas CLI

henn opened the pull request:
This address #354, and incorporates some of the functionality introduced in pull request #365.

zenhack commented on docs/USING.rst:
```diff
@@ -6,10 +6,22 @@ API are documented in `apidesc.md <apidesc.md>`_, and the mapping to HTTP is
+1. Set the ``HAAS_ENDPOINT`` environmental variable. An example (compatible
+   with the default development configuration) would be ``http://127.0.0.1:5000``
+
+   * If both configuration methods are present, this will take precedence over ``haas.cfg``.
```
I'd put this after you mention the other method.

zenhack commented on docs/USING.rst:
```diff
@@ -6,10 +6,22 @@ API are documented in `apidesc.md <apidesc.md>`_, and the mapping to HTTP is
+   with the default development configuration) would be ``http://127.0.0.1:5000``
+
+   * If both configuration methods are present, this will take precedence over ``haas.cfg``.
+   * The lack of a trailing slash can be significant.
```
Please expand on this.

zenhack commented on docs/USING.rst:
```diff
@@ -6,10 +6,22 @@ API are documented in `apidesc.md <apidesc.md>`_, and the mapping to HTTP is
+help`` will display an overview of the available commands. To tell ``haas``  which HaaS instance to use, be sure to do one of:
+
+1. Set the ``HAAS_ENDPOINT`` environmental variable. An example (compatible
+   with the default development configuration) would be ``http://127.0.0.1:5000``
```
There's nothing that's really stated as being the "default"; If you want to reference one of the example configs I would explicitly mention the file name.

zenhack commented on haas/cli.py:
```diff
@@ -66,7 +67,11 @@ def check_status_code(response):
-    url = cfg.get('client', 'endpoint')
+    # Prefer an environmental variable for getting the endpoint if available.
+    url = os.environ.get('HAAS_ENDPOINT')
+    if url == None:
```
[pep8 says][1] you should use `is` here:

`if url is None`

[1]: https://www.python.org/dev/peps/pep-0008/#id42

zenhack commented:
I made a few comments, but otherwise looks good to me.

zenhack commented:
Neat little trick for the future: if you put "resolves #bugnumber" in a commit message somewhere, then that issue will be closed automatically when the commit is merged into master.

It accepts a few other terms as well, e.g. closes or fixes

henn commented:
@zenhack I believe I've addressed all of your review concerns. I also rebased so that when merged these changes wouldn't bring with them all the micro-edits.

zenhack commented on docs/USING.rst:
```diff
@@ -6,10 +6,21 @@ API are documented in `apidesc.md <apidesc.md>`_, and the mapping to HTTP is
-serve_networks``, ``haas.cfg`` must be in your working directory. But here,
-only the ``client`` section is needed, so users without access to the global
-configuration file may use a more minimal one.
+help`` will display an overview of the available commands. To tell ``haas``  which HaaS instance to use, be sure to do one of:
```
Please hard-wrap this line to 80 chars.

zenhack commented on docs/USING.rst:
```diff
@@ -6,10 +6,21 @@ API are documented in `apidesc.md <apidesc.md>`_, and the mapping to HTTP is
+   endpoint = http://127.0.0.1:5000
+
+* If both configuration methods are present, the ``HAAS_ENDPOINT`` environmental variable will take precedence over whatever is contained within ``haas.cfg``.
+* Though insignificant in some circumstances, the presence or absence of trailing slashes within the endpoint URL can cause issues in communicating with the HaaS server. For example, using ``http://127.0.0.1:5000`` vs ``http://127.0.0.1:5000/``.
```
Same thing here; these two lines should be hard wrapped.

zenhack replied:
When I asked you to expand, I was thinking more in terms of "what kinds of issues could this cause?"

zenhack commented:
+1. @gsilvis, @lokI8, can one of you do a second review?

gsilvis commented:
Looks good.  +1 and merging now
